RELEASE-PROCEDURE.md

```diff
@@ -0,0 +1,49 @@
+c-ares release procedure - how to do a release
+==============================================
+
+in the source code repo
+-----------------------
+
+- edit `RELEASE-NOTES` to be accurate
+
+- make sure all relevant changes are committed on the master branch
+
+- tag the git repo in this style: `git tag -a cares-1_13_0` -a annotates the
+  tag and we use underscores instead of dots in the version number.
+
+- run "./maketgz 1.13.0" to build the release tarball. It is important that
+  you run this on a machine with the correct set of autotools etc installed
+  as this is what then will be shipped and used by most users on *nix like
+  systems.
+
+- push the git commits and the new tag
+
+- gpg sign the tarball
+
+- upload the resulting files to https://c-ares.haxx.se/download/
```

Comment:
Right now, those steps are restricted to me simply because that's how we've done in the past and because I haven't given anyone else an account to upload archives to the site.

I don't want to imply that we *have to* limit it to me for a release to get done. If I'd get run over by a bus tomorrow, you'd just upload them to another place until you'd sort out how to get access to the site or move the hosting to another machine.